src/primitives/transaction.h

```diff
@@ -460,12 +464,25 @@ struct CMutableTransaction
         SerializeTransaction(*this, s, ser_action);
     }
 
+    template <typename Stream>
+    CMutableTransaction(deserialize_t, Stream& s) {
+        Unserialize(s);
+    }
+
     /** Compute the hash of this CMutableTransaction. This is computed on the
      * fly, as opposed to GetHash() in CTransaction, which uses a cached result.
      */
     uint256 GetHash() const;
 };
 
+typedef std::shared_ptr<const CTransaction> CTransactionRef;
+static inline CTransactionRef MakeTransactionRef() { return std::make_shared<const CTransaction>(); }
+static inline CTransactionRef MakeTransactionRef(const CTransaction& txIn) { return std::make_shared<const CTransaction>(txIn); }
+static inline CTransactionRef MakeTransactionRef(const CMutableTransaction& txIn) { return std::make_shared<const CTransaction>(txIn); }
+static inline CTransactionRef MakeTransactionRef(CMutableTransaction&& txIn) { return std::make_shared<const CTransaction>(std::move(txIn)); }
```

Comment:
Might still be worth replacing the 6 functions with 3 using @theuni's suggestion